ceph/rados/core_workflows.py

```diff
@@ -1980,3 +1971,36 @@ def restart_daemon_services(self, daemon: str):
 
         log.info(f"Ceph Orch Service(s) {daemon_services} has been restarted")
         return True
+
+    def check_pool_pg_states(self, pool: str, disallowed_states: list):
+        """
+        the method fetches the states for the pg belonging to particular pool.
+        If any of the PG in the pool, have states that are in the disallowed states list, returns fail
+
+        Args::
+            pool: Name of the pool whose pg states need to be checked
+            disallowed_states: list of pg states, that are not allowed, if found, module returns fail
+
+        Returns::
+            Pass -> True
+            Fail -> False
+        """
+        # fetching the pool ID
+        cmd = "ceph df"
+        out = self.run_ceph_command(cmd=cmd)
+        pool_names = [entry["name"] for entry in out.get("pools", [])]
+        pool_id = out["pools"][pool_names.index(pool)]["id"]
+
+        cmd = f" ceph pg ls {pool_id}"
```

Comment:
If we already have the pool name, fetching the pool id is redundant, we can directly use `f"ceph pg ls-by-pool <pool_name>"`

Reply:
The command does not work with pool name. It requires pool ID.

Reply:
command for pool name - `f"ceph pg ls-by-pool <pool_name>"`